### PR TITLE
Display mucilage in specialized vacuole upgrade menu

### LINKED ISSUE #4589: Allow specialized vacuole to specialize for mucilage
Mucilage is stored in the normal storage so it doesn't make much sense why specialized vacuole can't be specialized for mucilage

### PR DESCRIPTION
**Pull request by CI09**
**Brief Description of What This PR Does**

Now code for displaying compounds in vacuole upgrade menu excludes agents except for mucilage.

Probably could have been done much better than that so I am open for suggestions.

**Related Issues**

Closes #4589

**Progress Checklist**

Note: before starting this checklist the PR should be marked as non-draft.

- [x] PR author has checked that this PR works as intended and doesn't
      break existing features:
      https://wiki.revolutionarygamesstudio.com/wiki/Testing_Checklist
      (this is important as to not waste the time of Thrive team
      members reviewing this PR)
- [ ] Initial code review passed (this and further items should not be checked by the PR author)
- [ ] Functionality is confirmed working by another person (see above checklist link)
- [ ] Final code review is passed and code conforms to the 
      [styleguide](https://github.com/Revolutionary-Games/Thrive/blob/master/doc/style_guide.md).

Before merging all CI jobs should finish on this PR without errors, if
there are automatically detected style issues they should be fixed by
the PR author. Merging must follow our
[styleguide](https://github.com/Revolutionary-Games/Thrive/blob/master/doc/style_guide.md#git).

### CODE REVIEW

File: src/microbe_stage/editor/upgrades/VacuoleUpgradeGUI.cs [hunk 1]
```diff
@@ -37,8 +37,17 @@ public override void _Ready()
 
     public void OnStartFor(OrganelleTemplate organelle, GameProperties currentGame)
     {
-        shownChoices = SimulationParameters.Instance.GetAllCompounds().Values
-            .Where(c => !c.IsEnvironmental && !c.IsAgent).ToList();
+        // Probably can be replaced with better logic
```
Comment (hhyyrylainen) [inline src/microbe_stage/editor/upgrades/VacuoleUpgradeGUI.cs:40]: Yeah, the upgrade GUI should have a private variable that holds the result of `SimulationParameters.Instance.GetCompound("mucilage")`. then the internal name of that variable can be compared with the thing that should be mucilage.

You can take a look at how for example the cell editor uses the sunlight variable:

https://github.com/Revolutionary-Games/Thrive/blob/master/src/microbe_stage/editor/CellEditorComponent.cs#L234

File: src/microbe_stage/editor/upgrades/VacuoleUpgradeGUI.cs [hunk 2]
```diff
@@ -37,8 +41,17 @@ public override void _Ready()
...
+        if (SimulationParameters.Instance.GetAllCompounds().ContainsKey("mucilage"))
```
Comment (hhyyrylainen) [inline src/microbe_stage/editor/upgrades/VacuoleUpgradeGUI.cs:45]: This if seems pointless and also the comment above should be fine to remove.

**Review by hhyyrylainen — APPROVED** [review]
The code change looks fine to me. I'll wait a bit before merging to see if anyone can playtest this.

**Comment by ender-titan1** [comment]
Can confirm. Everything works!

**Pull request merged**